cstore_reader.c

```diff
@@ -890,26 +1046,22 @@ DeserializeBoolArray(StringInfo boolArrayBuffer, uint32 boolArrayLength)
 			boolArray[boolArrayIndex] = true;
 		}
 	}
-
-	return boolArray;
 }
 
 
 /*
  * DeserializeDatumArray reads an array of datums from the given buffer and
- * returns the array. If a value is marked as false in the exists array, the
+ * stores them in provided datumArray. If a value is marked as false in the exists array, the
  * function assumes that the datum isn't in the buffer, and simply skips it.
  */
-static Datum *
-DeserializeDatumArray(StringInfo datumBuffer, bool *existsArray, uint32 datumCount,
+static void
+DeserializeDatumArray(StringInfo datumBuffer, bool *existsArray, Datum *datumArray, uint32 datumCount,
 					  bool datumTypeByValue, int datumTypeLength,
 					  char datumTypeAlign)
 {
```

Comment:
1. By looking at the function signature, the names ```existsArray``` and ```datumArray``` seem to be of same kind, but ```existsArray``` is an input argument and ```datumArray``` is a modify argument. I think we should make this more explicit. For example, we can rename ```datumArray``` to ```resultDatumArray```.
2. Input arguments and output arguments are mixed. I think we should either move all output arguments to the beginning of list or end of list. Please search our codes to see what we have used in other places.
3. First line exceeds the 85 character limit.